src/cljs/main/broadfcui/common/components.cljs

```diff
@@ -50,21 +50,23 @@
    :render
    (fn [{:keys [props]}]
      (let [{:keys [color icon href disabled? onClick text style class-name]} props]
-       [:a {:className (or class-name "button")
-            :style (merge
-                    {:display "inline-flex" :alignItems "center" :justifyContent "center"
-                     :backgroundColor (if disabled? (:disabled-state style/colors) color)
-                     :cursor (when disabled? "default")
-                     :color (if disabled? (:text-light style/colors) "white")
-                     :fontWeight 500
-                     :minHeight 19 :minWidth 19
-                     :borderRadius 2 :padding (if text "0.7em 1em" "0.4em")
-                     :textDecoration "none"}
-                    (if (map? style) style {}))
-            :href (or href "javascript:;")
-            :onClick (if disabled? (create-error-message disabled?) onClick)
-            :onKeyDown (when (and onClick (not disabled?))
-                         (common/create-key-handler [:space] onClick))}
+       [:a (merge
```

Comment:
I'm glad you asked!

I think when I first encountered (and worked around) this a while back, it was with a different component. Unlike `Button` and like `TextField`, that component used deeper composition before getting down to the actual (shadow) DOM element to apply the attributes to. Therefore, a new prop would have to be added to each layer going down. :(

Next, while we're using data-test-id at the moment, it may be useful to add additional data- props. The approach used here will not require further changes to the components to support additional data- props.

Finally, it seemed generally useful to me to be able to do this even beyond data- props. It could also be argued that the component should not allow this and instead pick and choose the props and decide how/where to apply them, in which case `TextField` should be changed. However, I was very happy with how easy it was to add the prop to `TextField` instances when I needed it.

When I asked @dmohs, he seemed to think that propagating the pattern used by `TextField` was good so that's what I did. I added you, @zarsky-broad, to this PR so you could present the counter opinion if you felt differently. Whatever we do is going to be done much more very soon, so let's make sure we make the right decision here.

Thanks.